modules/REST/Abstract_Controller.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST;
+
+use WP_Error;
+use WP_REST_Controller;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Abstract REST controller.
+ *
+ * @since 3.7
+ */
+abstract class Abstract_Controller extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param int $version Optional. Version of the API. Default is `1`.
+	 *
+	 * @phpstan-param positive-int $version
+	 */
+	public function __construct( int $version = 1 ) {
+		$this->namespace = "pll/v{$version}";
+	}
+
+	/**
+	 * Adds a status code to the given error and returns the error.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_Error $error       A `WP_Error` object.
+	 * @param int      $status_code Optional. A status code. Default is 400.
+	 * @return WP_Error
+	 */
+	protected function add_status_to_error( WP_Error $error, int $status_code = 400 ): WP_Error {
```

Comment:
If this all what languages and options share, a trait is sufficient? What do you think?

Reply:
I'm not sure to prefer a trait over an abstract class here, since we'll always extend `WP_REST_Controller` anyway.